engine/schema/src/main/java/com/cloud/capacity/dao/CapacityDao.java

```diff
@@ -24,8 +24,13 @@
 import com.cloud.utils.Pair;
 import com.cloud.utils.Ternary;
 import com.cloud.utils.db.GenericDao;
+import org.apache.cloudstack.framework.config.ConfigKey;
 
 public interface CapacityDao extends GenericDao<CapacityVO, Long> {
+
+    ConfigKey<Boolean> allowRoutersOnDedicatedResources =  new ConfigKey<>("Advanced", Boolean.class, "allow.routers.on.dedicated.resources", "false",
```

Comment:
I'm a bit weary having `Dao`s be `Configurable`s. Can this be setting be moved to the `Planner`-hierarchy somehow?